server/lib/notifications/agents/gotify.ts

```diff
@@ -0,0 +1,162 @@
+    type: Notification,
+    payload: NotificationPayload
+  ): {
+    title: string;
+    message: string;
+    priority: number;
+  } {
+    const { applicationUrl, applicationTitle } = getSettings().main;
+    let priority = 0;
+
+    const title = payload.event
+      ? `${payload.event} - ${payload.subject}`
+      : payload.subject;
+    let message = payload.message ?? '';
+
+    if (payload.request) {
+      message += `\n\nRequested By: ${payload.request.requestedBy.displayName}`;
+
+      let status = '';
+      switch (type) {
+        case Notification.MEDIA_PENDING:
+          status = 'Pending Approval';
+          break;
+        case Notification.MEDIA_APPROVED:
+        case Notification.MEDIA_AUTO_APPROVED:
+          status = 'Processing';
+          break;
+        case Notification.MEDIA_AVAILABLE:
+          status = 'Available';
+          break;
+        case Notification.MEDIA_DECLINED:
+          status = 'Declined';
+          break;
+        case Notification.MEDIA_FAILED:
+          status = 'Failed';
+          break;
+      }
+
+      if (status) {
+        message += `\nRequest Status: ${status}`;
+      }
+    } else if (payload.comment) {
+      message += `\nComment from ${payload.comment.user.displayName}:\n${payload.comment.message}`;
+    } else if (payload.issue) {
+      message += `\n\nReported By: ${payload.issue.createdBy.displayName}`;
+      message += `\nIssue Type: ${IssueTypeName[payload.issue.issueType]}`;
+      message += `\nIssue Status: ${
+        payload.issue.status === IssueStatus.OPEN ? 'Open' : 'Resolved'
+      }`;
+
+      if (type == Notification.ISSUE_CREATED) {
+        priority = 1;
+      }
+    }
+
+    for (const extra of payload.extra ?? []) {
+      message += `\n\n**${extra.name}**\n${extra.value}`;
+    }
+
+    if (applicationUrl && payload.media) {
+      const actionUrl = `${applicationUrl}/${payload.media.mediaType}/${payload.media.tmdbId}`;
+      message += `\n\nOpen in ${applicationTitle}(${actionUrl})`;
+    }
+
+    return {
+      title,
+      message,
+      priority,
+    };
+  }
+
+  public async send(
+    type: Notification,
+    payload: NotificationPayload
+  ): Promise<boolean> {
+    const settings = this.getSettings();
+
+    if (!hasNotificationType(type, settings.types ?? 0)) {
+      return true;
+    }
+
+    logger.debug('Sending Gotify notification', {
+      label: 'Notifications',
+      type: Notification[type],
+      subject: payload.subject,
+    });
+    try {
+      const endpoint = `${settings.options.url}/message?token=${settings.options.token}`;
+
+      const { title, message, priority } = this.constructMessageDetails(
+        type,
+        payload
+      );
+
+      await axios.post(endpoint, {
+        extras: {
+          'client::display': {
+            contentType: 'text/markdown',
+          },
+        },
+        title: title,
+        message: message,
+        priority: priority,
+      } as GotifyPayload);
```

Comment:
Why not just construct the `GotifyPayload` in a `getNotificationPayload()` method?  See Pushbullet agent for example.

Reply:
implemented suggested change in 4fcdf